cluster/examples/kubernetes/ceph/psp/kube-registry.yaml

```diff
@@ -0,0 +1,129 @@
+# Example showing how to run the kube-registry application with pod security policies in effect.
+# Attempts have been made to make sure the policy, role, and binding in this example follow
+# best practices and are a minimal set of permissions; however, this is not a guarantee of a
+# minimal policy config.
+---
+# Create a service account for the kube registry
+# Limit it to the kube-system namespace
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kube-registry
+  namespace: kube-system
+---
+# Create a pod security policy  (psp) for the service account
+# that is as minimal as possible.
+# TODO: This could probably be more minimal
+apiVersion: extensions/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: kube-registry.psp
+spec:
+  fsGroup:
+    rule: RunAsAny
+  privileged: false
+  runAsUser:
+    rule: RunAsAny
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  volumes:
+  - flexVolume
+  - secret
+  allowedFlexVolumes:
+  - driver: "ceph.rook.io/rook"
+  - driver: "ceph.rook.io/rook-ceph"
+  hostPID: false
+  hostIPC: false
+  hostNetwork: false
+---
+# Create a role for the kube-registry that can use
+# the security policy
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: Role
+metadata:
+  name: kube-registry:psp
+  namespace: kube-system
+rules:
+- apiGroups:
+  - extensions
+  resources:
+  - podsecuritypolicies
+  resourceNames:
+  - kube-registry.psp
+  verbs:
+  - use
+---
+# Bind the role to the service account so that all pods
+# created by controllers in in the service account can
+# create the resources allowed by the psp
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: kube-registry:psp
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kube-registry:psp
+subjects:
+- kind: ServiceAccount
+  name: kube-registry
+  namespace: kube-system
+---
+# Create the HA registry
+apiVersion: v1
```

Comment:
is this replication controller the same as in kube-registry.yaml, except for the service account?

Reply:
Yes except that the pod spec specifies that it runs with the `kube-registry` service account. I could potentially add that service account to `kube-registry.yaml`, and then have a separate `kube-registry-psp.yaml` file, and then I don't think there would be duplication.